packages/pegasus/src/courier.js

```diff
@@ -0,0 +1,124 @@
+import { details as X } from '@agoric/assert';
+
+import { AmountMath } from '@agoric/ertp';
+import { E } from '@agoric/eventual-send';
+import { Far } from '@agoric/marshal';
+import { makeOncePromiseKit } from './once-promise-kit.js';
+
+/**
+ * Create or return an existing courier promise kit.
+ *
+ * @template K
+ * @param {K} key
+ * @param {Store<K, PromiseRecord<Courier>>} keyToCourierPK
+ */
+export const getCourierPK = (key, keyToCourierPK) => {
+  if (keyToCourierPK.has(key)) {
+    return keyToCourierPK.get(key);
+  }
+
+  // This is the first packet for this denomination.
+  // Create a new Courier promise kit for it.
+  const courierPK = makeOncePromiseKit(() => X`${key} already pegged`);
```

Comment:
Consider simplifying to
```suggestion
  const courierPK = makeOncePromiseKit(X`${key} already pegged`);
```
I understand that this *may* be more expensive than thunkifying it. But `X` was designed to be pretty cheap. If it is cheap enough, I'd prefer the simpler code.